### PR TITLE
core: remove deprecated UID methods

### PR DESCRIPTION
**Pull request by julianoes**
This removes the old UUID stuff which has been marked deprecated for a while. The new way to access the uid is using the Info plugin.

### CODE REVIEW

**Review by TSC21 — DISMISSED** [review]

File: src/integration_tests/action_transition_multicopter_fixedwing.cpp [hunk 1]
```diff
@@ -44,11 +33,35 @@ void takeoff_and_transition_to_fixedwing()
...
+    ASSERT_EQ(action_ret, Action::Result::Success);
 
     LogInfo() << "Waiting until hovering";
-    wait_until_hovering(telemetry);
+    auto prom = std::promise<void>{};
+    auto fut = prom.get_future();
+    // Wait until hovering.
+    telemetry->subscribe_flight_mode([&telemetry, &prom](Telemetry::FlightMode mode) {
+        if (mode == Telemetry::FlightMode::Hold) {
+            telemetry->subscribe_flight_mode(nullllptr);
```
Comment (JonasVautherin) [inline src/integration_tests/action_transition_multicopter_fixedwing.cpp:59]: `nullllptr`? :sweat_smile:

**Review by JonasVautherin — CHANGES_REQUESTED** [review]

File: src/integration_tests/action_transition_multicopter_fixedwing.cpp (hunk 1 above)
Comment (julianoes) [reply src/integration_tests/action_transition_multicopter_fixedwing.cpp:59]: Haha, that must have slipped in after testing.

Comment (julianoes) [reply src/integration_tests/action_transition_multicopter_fixedwing.cpp:59]: Ah yes, I remember typing `all` for `git add --all` and it was in the wrong window but I couldn't see where it went.

Comment (julianoes) [reply src/integration_tests/action_transition_multicopter_fixedwing.cpp]: Also, beware of `nullllptr`, they are extra null, at least double!

**Review by JonasVautherin — APPROVED** [review]

**Pull request merged**